Makefile

```diff
@@ -22,30 +22,66 @@ cgo_cflags :=
 cgo_ldflags :=
 cc :=
 
-uname_s := $(shell uname -s)
-ifeq ($(uname_s),Darwin)
+# Build platform flags
+BUILDOS ?= $(shell uname -s | tr '[:upper:]' '[:lower:]')
+BUILDARCH ?= $(shell uname -m | tr '[:upper:]' '[:lower:]')
+ifeq ($(BUILDARCH),aarch64)
+        BUILDARCH=arm64
+endif
+ifeq ($(BUILDARCH),x86_64)
+        BUILDARCH=amd64
+endif
+
+# Override these (not BUILDOS/BUILDARCH) for cross-compilation
+export GOOS ?= $(BUILDOS)
+export GOARCH ?= $(BUILDARCH)
+
+# Cross-compilation section. Currently supported:
+# darwin amd64 --> darwin arm64
+# darwin arm64 --> linux arm64
+# linux amd64 --> linux arm64
+# linux amd64 --> windows amd64
+
+ifeq ($(BUILDOS),darwin)
+	ifeq ($(GOOS),darwin)
 		cgo_ldflags += -framework CoreFoundation -framework Security
-	ifeq ($(GOARCH),arm64)
-		cgo_cflags += --target=arm64-apple-macos11
-		cgo_ldflags += --target=arm64-apple-macos11
+		ifeq ($(BUILDARCH),amd64)
+			ifeq ($(GOARCH),arm64)
+				cgo_cflags += --target=arm64-apple-macos11
+				cgo_ldflags += --target=arm64-apple-macos11
+			endif
+		endif
 	endif
-endif
 
-ifeq ($(uname_s),Linux)
-	ifeq ($(GOARCH),arm64)
-		cgo_cflags += --target=aarch64-linux-gnu
-		cgo_ldflags += --target=aarch64-linux-gnu
-		cc = clang --sysroot=/usr/aarch64-linux-gnu
+	ifeq ($(GOOS),linux)
+		ifeq ($(GOARCH),arm64)
+			LLVM_PATH ?= /opt/homebrew/opt/llvm/bin
+			SYSROOT ?= /tmp/sysroot-aarch64-linux-gnu
+			cgo_cflags += --target=aarch64-linux-gnu -Wno-error=unused-command-line-argument
```

Comment:
I swear I tried 100 different methods and nothing I tried was able to get `-fuse-ld=$(LLVM_PATH)/ld.lld` passed to `clang` when it was running in linker mode consistently. So now we pass it to _every_ `clang` command, and `-Wno-error=unused-command-line-argument` just ignores it when we're compiling rather than linking. I don't know. There's probably a better way, but no combination of `-ldflags`, `-extld`, `-extldflags`, `LDFLAGS`, and a few others was making it happen for me.